tests/snapshot/terraform/expressions_functions.tf.snap

```diff
@@ -356,4 +356,29 @@
 #     ^^^ source.hcl.terraform meta.function-call.hcl string.quoted.double.hcl
 #        ^ source.hcl.terraform meta.function-call.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
 #         ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.end.hcl
+>
+># provider defined functions
+#^ source.hcl.terraform comment.line.number-sign.hcl punctuation.definition.comment.hcl
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform comment.line.number-sign.hcl
+>
+>provider::framework::example("hi")
+#^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.hcl.terraform meta.function-call.hcl support.function.provider
+#                            ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.begin.hcl
+#                             ^ source.hcl.terraform meta.function-call.hcl string.quoted.double.hcl punctuation.definition.string.begin.hcl
+#                              ^^ source.hcl.terraform meta.function-call.hcl string.quoted.double.hcl
+#                                ^ source.hcl.terraform meta.function-call.hcl string.quoted.double.hcl punctuation.definition.string.end.hcl
+#                                 ^ source.hcl.terraform meta.function-call.hcl punctuation.section.parens.end.hcl
+>invalid::namespaced::function("bye")
+#^^^^^^^ source.hcl.terraform
+#       ^ source.hcl.terraform
+#        ^ source.hcl.terraform
+#         ^^^^^^^^^^ source.hcl.terraform
+#                   ^ source.hcl.terraform
+#                    ^ source.hcl.terraform
+#                     ^^^^^^^^ source.hcl.terraform meta.function-call.hcl
```

Comment:
I'm a bit unsure about this still identifying a function call – should I e.g. add a negative look-ahead to the non-namespaced function regex so that `meta-function-call.hcl` is not matched anymore?
Or does this overly complicate things for code that's invalid anyway?

Reply:
You could take a quick look at this if it's not too much effort. To me, syntax highlighting should be primarily about highlighting valid configuration correctly.

Reply:
Had a quick look, but it would require splitting up the start regex and it would complicate things quite a bit, so I opted for not fixing this